trie/sync.go

```diff
@@ -71,6 +71,20 @@ type request struct {
 //   - Path 0x012345678901234567890123456789010123456789012345678901234567890199 -> {0x0123456789012345678901234567890101234567890123456789012345678901, 0x0099}
 type SyncPath [][]byte
 
+// newSyncPath converts an expanded trie path from nibble form into a compact
+// version that can be sent over the network.
+func newSyncPath(path []byte) SyncPath {
+	// If the hash is from the account trie, append a single item, if it
+	// is from the a storage trie, append a tuple. Note, the length 64 is
+	// clashing between account leaf and storage root. It's fine thpough
+	// because having a trie node at 64 depth means a hash collition was
```

Comment:
```suggestion
	// because having a trie node at 64 depth means a hash colliison was
```
Unless you mean a hash `coalition`, which is less violent and more civilized gathering of hashes...